config/management/operational/src/governance.rs

```diff
@@ -2,11 +2,104 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{json_rpc::JsonRpcClientWrapper, TransactionContext};
+use libra_crypto::ed25519::Ed25519PublicKey;
 use libra_global_constants::LIBRA_ROOT_KEY;
-use libra_management::{config::Config, error::Error, transaction::build_raw_transaction};
-use libra_types::{account_address::AccountAddress, account_config::libra_root_address};
+use libra_management::{
+    config::{Config, ConfigPath},
+    error::Error,
+    transaction::build_raw_transaction,
+};
+use libra_types::{
+    account_address::AccountAddress,
+    account_config::libra_root_address,
+    chain_id::ChainId,
+    transaction::{authenticator::AuthenticationKey, Script},
+};
+use std::path::PathBuf;
 use structopt::StructOpt;
 
+#[derive(Debug, StructOpt)]
+pub struct CreateAccount {
+    #[structopt(flatten)]
+    config: ConfigPath,
+    #[structopt(long)]
+    name: String,
+    #[structopt(long)]
+    path_to_key: PathBuf,
+    #[structopt(long, required_unless = "config")]
+    json_server: Option<String>,
+    #[structopt(long, required_unless("config"))]
+    chain_id: Option<ChainId>,
+}
+
+impl CreateAccount {
+    fn config(&self) -> Result<Config, Error> {
+        self.config.load().map(|c| {
+            c.override_chain_id(self.chain_id)
+                .override_json_server(&self.json_server)
+        })
+    }
+
+    fn key(&self) -> Result<Ed25519PublicKey, Error> {
+        libra_management::read_key_from_file(&self.path_to_key)
+            .map_err(|e| Error::UnableToReadFile(format!("{:?}", self.path_to_key), e))
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct CreateValidator {
+    #[structopt(flatten)]
+    input: CreateAccount,
+}
+
+impl CreateValidator {
+    pub fn execute(self) -> Result<(TransactionContext, AccountAddress), Error> {
+        let config = self.input.config()?;
+        let key = self.input.key()?;
+        let client = JsonRpcClientWrapper::new(config.json_server.clone());
+
+        let seq_num = client.sequence_number(libra_root_address())?;
+        let auth_key = AuthenticationKey::ed25519(&key);
+        let account_address = auth_key.derived_address();
+        let script = transaction_builder::encode_create_validator_account_script(
```

Comment:
This flow diverges from the one in genesis (where we also allow auth keys to be none in the case that the owner doesn't provide a key..). Do we want to support this functionality too here, or just leave it as is and use a throw away key?

Reply:
The problem is what is exposed via the public interfaces versus the admin scripts. This is the best we can do with public interfaces. It most certainly does not match what genesis will do whereas other systems will. We can follow up later to explore admin scripts to generate the account, but definitely not critical for e2e tests.

Reply:
Sure, that makes sense! Just wanted to bring it up to make sure we're aware of the divergence 😄